server/src-lib/Hasura/Server/Query.hs

```diff
@@ -259,10 +259,10 @@ queryModifiesSchemaCache (RQV1 qi) = case qi of
   RQAddCollectionToAllowlist _    -> True
   RQDropCollectionFromAllowlist _ -> True
 
-  RQRunSql RunSQL{rTxAccessMode}  ->
+  RQRunSql RunSQL{rTxAccessMode, rCheckMetadataConsistency}  ->
     case rTxAccessMode of
       Q.ReadOnly  -> False
-      Q.ReadWrite -> True
+      Q.ReadWrite -> maybe True id rCheckMetadataConsistency
```

Comment:
`maybe x id` can (and should!) be abbreviated to `fromMaybe x`. I’ll make the change here myself, since it’s such a tiny thing, but just so you know!

Reply:
Oh, I didn't know about `fromMaybe`. Thanks for the fix and the suggestion 🙂